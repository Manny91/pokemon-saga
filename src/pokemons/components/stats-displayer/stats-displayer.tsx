import React from "react";
import styled from "../../../styled-components";
import { PokemonStat } from "../../../services/pokemon.service";

interface PokemonStatsDisplayerProps {
  stats: PokemonStat[];
}
export function PokemonStatsDisplayer({ stats }: PokemonStatsDisplayerProps) {
  return (
    stats && (
      <StatsWrapper>
        {stats.map((stat, i) => {
          return (
            <PokeStatContainer key={i}>{statToString(stat)}</PokeStatContainer>
          );
        })}
      </StatsWrapper>
    )
  );
}

function statToString({ stat, base_stat }: PokemonStat): string {
  if (stat) {
    const maxLengthString = 20;
    const statName = stat.name;
    const pokeStatStringLength = statName.length + base_stat.toString().length;
    return (
      statName +
      new Array(maxLengthString - pokeStatStringLength).fill(".").join("") +
      base_stat
    );
  }
  return "";
}

const StatsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-grow: 1;
  background-color: ${props => props.theme.colors.greenScreen};
  padding: 15px 20px;
  border-radius: 3px;
  border: inset ${props => props.theme.colors.greenBorder} 3px;
`;

const PokeStatContainer = styled.p`
  margin: 5px 0px;
  font-weight: bold;
  text-transform: capitalize;
`;
