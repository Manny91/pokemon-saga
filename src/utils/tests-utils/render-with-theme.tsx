import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
