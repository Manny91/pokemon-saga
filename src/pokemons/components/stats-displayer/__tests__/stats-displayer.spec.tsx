import React from "react";
import { PokemonStatsDisplayer } from "../stats-displayer";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";

describe("PokemonStatsDisplayer Component", () => {
  it("should render properly", () => {
    const testProps = {
      stats: [
        {
          base_stat: 70,
          effort: 0,
          stat: { name: "speed" }
        }
      ]
    };
    const subject = renderWithTheme(<PokemonStatsDisplayer {...testProps} />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
