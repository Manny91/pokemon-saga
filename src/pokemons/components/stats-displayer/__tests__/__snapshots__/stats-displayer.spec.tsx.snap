// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonStatsDisplayer Component should render properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  background-color: #A5CD53;
  padding: 15px 20px;
  border-radius: 3px;
  margin: 5px;
  border: inset #9aa28b 3px;
}

.c1 {
  margin: 5px 0px;
  font-weight: bold;
  text-transform: capitalize;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
    >
      speed.............70
    </p>
  </div>
</div>
`;
